prov/shm/src/smr_progress.c

```diff
@@ -66,6 +66,7 @@ static int smr_progress_resp_entry(struct smr_ep *ep, struct smr_resp *resp,
 	struct smr_inject_buf *tx_buf = NULL;
 	struct smr_sar_msg *sar_msg = NULL;
 	uint8_t *src;
+	bool locked = false;
```

Comment:
You can drop this now and just repeat the pointer comparison.  I would add a short comment to the code regarding why the lock isn't required as well.